src/main/java/mil/dds/anet/database/PositionDao.java

```diff
@@ -170,134 +167,204 @@ public int updateInternal(Position p) {
-          + " :createdAt)";
-    } else {
-      insertSql = "/* positionRemovePerson.insert1 */ INSERT INTO \"peoplePositions\" "
-          + "(\"positionUuid\", \"personUuid\", \"createdAt\") VALUES(NULL, "
-          + "(SELECT \"personUuid\" FROM \"peoplePositions\""
-          + " WHERE \"positionUuid\" = :positionUuid ORDER BY \"createdAt\" DESC LIMIT 1),"
-          + " :createdAt)";
-    }
-    getDbHandle().createUpdate(insertSql).bind("positionUuid", positionUuid)
-        .bind("createdAt", DaoUtils.asLocalDateTime(now)).execute();
+        .bind("updatedAt", DaoUtils.asLocalDateTime(now)).bind("positionUuid", positionUuid)
+        .execute();
+  }
 
+  @InTransaction
+  public int insertPersonEmptyInPosition(String positionUuid) {
+    Instant now = Instant.now();
     return getDbHandle()
-        .createUpdate("/* positionRemovePerson.insert2 */ INSERT INTO \"peoplePositions\" "
+        .createUpdate("/* positionEmptyPerson.insert */ INSERT INTO \"peoplePositions\" "
             + "(\"positionUuid\", \"personUuid\", \"createdAt\") "
             + "VALUES (:positionUuid, NULL, :createdAt)")
         .bind("positionUuid", positionUuid)
         // Need to ensure this timestamp is greater than previous INSERT.
         .bind("createdAt", DaoUtils.asLocalDateTime(now.plusMillis(1))).execute();
   }
 
+  @InTransaction
+  public int insertPositionEmptyInPerson(String personUuid) {
+    Instant now = Instant.now();
+    return getDbHandle()
+        .createUpdate("/* positionEmptyPerson.insert */ INSERT INTO \"peoplePositions\" "
+            + "(\"positionUuid\", \"personUuid\", \"createdAt\") "
+            + "VALUES (NULL, :personUuid, :createdAt)")
+        .bind("personUuid", personUuid)
+        // Need to ensure this timestamp is greater than previous INSERT.
+        .bind("createdAt", DaoUtils.asLocalDateTime(now.plusMillis(1))).execute();
+  }
+
+  private int deletePersonEmptyInPositionEmpty() {
+    return getDbHandle()
+        .createUpdate("/* positionEmptyPersonAndPosition.delete */ DELETE FROM \"peoplePositions\" "
+            + " WHERE \"personUuid\" IS NULL and \"positionUuid\" IS NULL")
+        .execute();
+  }
+
+  @InTransaction
+  public int removePersonFromPosition(String positionUuid) {
+    int numRows = 0;
+    Instant now = Instant.now();
+    numRows += cleanPosition(positionUuid);
+
+    PersonPositionHistory currentPositionPerson = getDbHandle()
+        .createQuery("/* personEmptyPosition.find */ SELECT * FROM \"peoplePositions\""
+            + " WHERE \"positionUuid\" = :positionUuid AND \"personUuid\" IS NOT NULL AND "
+            + " \"endedAt\" IS NULL")
+        .bind("positionUuid", positionUuid).map(new PersonPositionHistoryMapper()).findFirst()
+        .orElse(null);
+
+    numRows += getDbHandle()
+        .createUpdate(
+            "/* positionRemovePerson.end */ UPDATE \"peoplePositions\" SET \"endedAt\" = :endedAt "
```

Comment:
Please ensure that each database statement has a unique comment.

Reply:
Fixed with using unique comments.